File: SeongYeon/js/programmers/visitLength.js
function solution(dirs) {
  const move = { U: [-1, 0], D: [1, 0], L: [0, -1], R: [0, 1] };
  const passed = new Set();
  let row = 0,
    col = 0;

  for (dir of dirs) {
    const prevRow = row;
    const prevCol = col;

    row += move[dir][0];
    col += move[dir][1];

    if (row < -5 || row > 5 || col < -5 || col > 5) {
      row = prevRow;
      col = prevCol;
      continue;
    }

    passed.add(`${prevRow}${prevCol}-${row}${col}`);
    passed.add(`${row}${col}-${prevRow}${prevCol}`);
  }

  return passed.size / 2;
}

File: JiHwan/BJ/visit_length.py
def solution(dirs):
    x = 0
    y = 0
    result = set()
    direction = {'U': (0, 1), 'D': (0, -1), 'L': (-1, 0), 'R': (1, 0)}

    for d in dirs:
        dx, dy = direction[d]
        nx = x + dx
        ny = y + dy

        if -5 <= nx <= 5 and -5 <= ny <= 5:
            result.add(((x, y), (nx, ny)))
            result.add(((nx, ny), (x, y)))
            x = nx
            y = ny
        # print(result)
    return len(result) // 2